p2p/peer.py

```diff
@@ -955,7 +955,11 @@ def _add_peer(self,
                 break
 
             start_block = vm_class.dao_fork_block_number - 1
-            request = peer.request_block_headers(start_block, max_headers=2)  # type: ignore
+            request = cast(ETHPeer, peer).request_block_headers(
```

Comment:
This could be a `LESPeer` as well. Not sure if the `# type ignore` comment is any better, but the cast is misleading

Reply:
I updated to use an `ignore` comment and added a comment indicating that this *can* be an `LESPeer` and that this issue will be fixed when the full awaitable round-trip API is in place.